### DISCUSSION TITLE
Ratpack requires manual instrumentation?

### DISCUSSION THREAD

**Discussion by johnrengelman** [open-telemetry/opentelemetry-java-instrumentation#12507] [Q&A]
Hi all - 
I've recently been implementing otel metrics across some Ratpack apps and we are using a combination of the javaagent and the manual instrumentation. As I've dug into it, I'm trying to understand why it's necessary to do the manual instrumentation? Is it not possible to use the same AOP injection methods to get these components in place? Is it a matter of wanting to allow some user control on configuration?

Just wanted to get some more info before I dig in deeper.

**Comment by trask** [comment]
hi @johnrengelman!

it's probably nothing intentional, more likely just no one has reported a specific issue about it, I'm not sure how heavily the ratpack instrumentation is used, though we did have a recent issue and improvement worth looking at: https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/12317, and certainly issues and/or PRs would be welcome if you find ways to improve it further

**Reply by johnrengelman** [reply]
Thanks for that @trask. I'll see if I can take a look at integrating some of this into the agent to inject itself. I see a similar pattern in the `netty` implementations that facilitates both manual and automatic instrumentation.

I've also noticed some issues where the Netty and Ratpack instrumentation is colliding. Specifically the context for a Ratpack Http Client request doesn't become the parent for the netty context for the outgoing connection.
![2024-10-29 at 2 35 PM](https://github.com/user-attachments/assets/0ec3e6ed-7b47-465f-9a8a-2595e6f3686e)
I suspect this is because Ratpack doesn't have a way of shoving the context down into the channel as an attribute.
So i've been turning off the netty instrumentation in these cases, but wondering if there is a better way of doing this in the absence of a fix.
